mono/tests/pinvoke2.cs

```diff
@@ -1088,6 +1088,9 @@ class C {
 		catch (ArgumentException) {
 		}
 
+		if (mono_test_asany (new IntPtr (5), 1) != 0)
```

Comment:
I think there's an endianness problem here. The C code for this case does `*(int*)arg` so for a pointer-sized argument it's going to see the wrong 32-bits on a BE platform.